.github/workflows/deploy_infra.yml

```diff
@@ -0,0 +1,146 @@
+      ORG_GITHUB_PACKAGES_READ_ONLY_TOKEN:
+        required: true
+      CORE_ENG_DEPLOYMENTS_SLACK_WEBHOOK:
+        required: true
+
+concurrency:
+  group: ${{ github.workflow }}-${{ inputs.environment }}
+
+permissions:
+  id-token: write
+  contents: read
+
+env:
+  TERRAFORM_ROOT: ${{ inputs.environment }}
+  TERRAFORM_WORKSPACE: ${{ inputs.workspace }}
+  TF_INPUT: false
+  TF_IN_AUTOMATION: true
+
+jobs:
+  Setup:
+    runs-on: ubuntu-latest
+    outputs:
+      commit-tag: ${{ steps.get-commit-tag.outputs.commit-tag }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4.1.6
+        with:
+          ref: ${{ inputs.commit-identifier }}
+
+      - name: Get Build Commit Tag
+        id: get-commit-tag
+        run: |
+          echo "commit-tag=$(git rev-parse --short HEAD)" >> $GITHUB_OUTPUT
+
+      - name: Markdown Summary
+        run: |
+          echo "### Deploy Infrastructure Context"  >> $GITHUB_STEP_SUMMARY
+          echo "**Trigger**: ${{ github.event_name }} ${{ github.event.action }}" >> $GITHUB_STEP_SUMMARY
+          echo "**Actor**: ${{ github.actor }}" >> $GITHUB_STEP_SUMMARY
+          echo "**Workflow Ref (branch)**: ${{ github.workflow_ref }}" >> $GITHUB_STEP_SUMMARY
+          echo -e "**Workflow Sha (commit)**: ${{ github.workflow_sha }}\n" >> $GITHUB_STEP_SUMMARY
+          echo "**Inputs**" >> $GITHUB_STEP_SUMMARY
+          echo -e "${{ toJson(inputs) }}\n" >> $GITHUB_STEP_SUMMARY
+          echo "**Release Tag**: ${{ inputs.release-tag }}" >> $GITHUB_STEP_SUMMARY
+          echo "**Commit Tag**: ${{ steps.get-commit-tag.outputs.commit-tag }}" >> $GITHUB_STEP_SUMMARY
+          echo "*Note: Workflow Ref & Sha represent the branch and commit of the workflow defintion. This may not be the same as the branch/commit checked out supplying the code. The 'SHA or tag of tests to run' aka commit-identifier, is what specifies the test defintions. If a commit-identifier is not supplied, the head of the selected branch is used.*" >> $GITHUB_STEP_SUMMARY
+
+      - name: Validate Environment
+        run: |
+          if [[ ! "${{ inputs.environment }}" =~ ^(dev|stage|prod)$ ]]; then
+            echo "Invalid environment: ${{ inputs.environment }}"
+            exit 1
+          fi
+
+  Terraform-Apply:
+    name: Terraform Apply
+    runs-on: ubuntu-latest
+    environment: ${{ inputs.environment}}
+    defaults:
+      run:
+        working-directory: terraform/${{ inputs.environment }}
+
+    steps:
+      - id: get-role-arn
+        uses: OpenSesame/gha-oidc-access/get-role-arn@v2
+        with:
+          domain: core
+          env: ${{ inputs.environment }}
+          ORG_READ_ONLY_SSH_KEY: ${{ secrets.ORG_READ_ONLY_SSH_KEY }}
+      
+      - uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ steps.get-role-arn.outputs.role-arn }}
+          role-session-name: ${{ inputs.workspace }}-${{ inputs.environment }}-Run${{ github.run_id }}
```

Comment:
sometimes this will evaluate to dev-dev-run01234

let's try using {{ github.repository }} in place of workspace and see if uses the outer repo from the calling workflow (I think it will, fingers crossed)

Reply:
It will I'm pretty sure, path and working directory also use the relative repository where the initial workflow  is defined and executed

Reply:
It will I'm pretty sure, path and working directory also use the relative repository where the initial workflow is defined and executed